clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java

```diff
@@ -162,6 +200,24 @@ public ListTopicsResult listTopics() {
     /**
      * List the topics available in the cluster.
      *
+     * <p>The following exceptions can be anticipated when calling {@code get()} on the futures obtained from the
+     * {@link ListTopicsResult#namesToListings() namesToListings()}, {@link ListTopicsResult#listings() listings()},
+     * {@link ListTopicsResult#names() names()} methods of the  returned {@code ListTopicsResult}</p>
+     *
+     * <ul>
+     *     <li>{@link org.apache.kafka.common.errors.UnknownTopicOrPartitionException}
+     *     if the authorized topic was in metadata first, but was not found during later look up</li>
```

Comment:
We don't use UnknownTopicOrPartitionException in listTopics

Reply:
When I was doing the research, it was there https://github.com/adyach/kafka/blob/KAFKA-5445/core/src/main/scala/kafka/server/KafkaApis.scala#L952

Reply:
but it is in https://github.com/apache/kafka/blob/88307657dff0202804447d06eef6ecdc808590d8/core/src/main/scala/kafka/server/KafkaApis.scala#L935

Reply:
That line is not there any more.

In any case, it doesn't really make sense to return this to the user.  list topics is supposed to list all topics.  What would returning UnknownTopicOrPartition mean in this context?

Reply:
@cmccabe I have removed from the doc 7a42286f1, but it is still in the code, so it could be returned to the user of that class.